x-pack/platform/plugins/shared/ml/public/application/model_management/trained_models_service.ts

```diff
@@ -0,0 +1,643 @@
+            withLatestFrom(this._scheduledDeployments$),
+            filter(([deployment, scheduled]) =>
+              scheduled.some(
+                (d) =>
+                  d.deploymentParams.deployment_id === deployment.deploymentParams.deployment_id
+              )
+            ),
+            // Now, separately check if a similar deployment already exists.
+            tap(([deployment]) => {
+              if (this.isDeploymentAlreadyExists(deployment, modelItems)) {
+                this.removeScheduledDeployment(deployment);
+              }
+            }),
+            // Only allow those deployments that still exist and were not removed.
+            filter(([deployment, scheduled]) =>
+              scheduled.some(
+                (d) =>
+                  d.deploymentParams.deployment_id === deployment.deploymentParams.deployment_id
+              )
+            ),
+            map(([deployment]) => deployment)
+          )
+        ),
+        tap((deployment) => this.executeScheduledDeployment(deployment))
+      )
+      .subscribe();
+
+    this.subscription.add(deploymentsSub);
+  }
+
+  private executeScheduledDeployment(deployment: ModelDeploymentParams) {
+    this.deploymentInProgress.add(deployment.modelId);
+
+    const singleDeploymentSub = this.getModel$(deployment.modelId)
+      .pipe(
+        filter((model) => this.isModelReadyForDeployment(model)),
+        take(1),
+        tap(() => this.updateUiStateForDeployment(deployment.modelId)),
+        switchMap(() => {
+          return from(
+            this.trainedModelsApiService.startModelAllocation(
+              deployment.modelId,
+              deployment.deploymentParams,
+              deployment.adaptiveAllocationsParams
+            )
+          );
+        }),
+        tap({
+          next: () => {
+            this.displaySuccessToast?.({
+              title: i18n.translate('xpack.ml.trainedModels.modelsList.startSuccess', {
+                defaultMessage: 'Deployment started',
+              }),
+              text: i18n.translate('xpack.ml.trainedModels.modelsList.startSuccessText', {
+                defaultMessage: '"{deploymentId}" has started successfully.',
+                values: { deploymentId: deployment.deploymentParams.deployment_id },
+              }),
+            });
+          },
+          error: (error) => {
+            this.displayErrorToast?.(
+              error,
+              i18n.translate('xpack.ml.trainedModels.modelsList.startFailed', {
+                defaultMessage: 'Failed to start "{deploymentId}"',
+                values: { deploymentId: deployment.deploymentParams.deployment_id },
+              })
+            );
+          },
+        }),
+        finalize(() => {
+          this.removeScheduledDeployment(deployment);
+          // Manually update the BehaviorSubject to ensure proper cleanup
+          // if user navigates away, as localStorage hook won't be available to handle updates
```

Comment:
It means that local storage service is necessary to avoid this. Definitely deserves an issue